src/main/clojure/clara/rules/engine.cljc

```diff
@@ -646,8 +646,8 @@
       (into [type] constraints))))
 
 (defn- join-node-matches
-  [node join-filter-fn token fact env]
-  (let [beta-bindings (try (join-filter-fn token fact {})
+  [node join-filter-fn token fact fact-bindings env]
+  (let [beta-bindings (try (join-filter-fn token fact fact-bindings {})
                            (catch #?(:clj Exception :cljs :default) e
                                (throw-condition-exception {:cause e
```

Comment:
Makes sense. I'm more tempted to go with token-bindings and fact-bindings here so we keep the distinction. 

Technically this could break a user explicitly catching an exception and checking the :bindings field, but I think that's low enough risk that it's worth using more clear names.